include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -448,42 +451,64 @@ TridiagResult<T, Device::CPU> BandToTridiag<Backend::MC, D, T>::call_L(
       dlaf::internal::whenAllLift(start, std::min(nb, size - start), std::min(nb, size - 1 - start),
                                   mat_trid.readwrite_sender(GlobalTileIndex{0, tile_index}),
                                   std::forward<decltype(dep)>(dep)) |
-          dlaf::internal::transformDetach(policy_hp, copy_tridiag_task);
+          dlaf::internal::transformDetach(policy, copy_tridiag_task);
+    }
+    else {
+      ex::start_detached(std::forward<decltype(dep)>(dep));
     }
   };
 
-  const auto sweeps = nrSweeps<T>(size);
-  for (SizeType sweep = 0; sweep < sweeps; ++sweep) {
-    // Create the first max_workers workers and then reuse them.
-    auto& w_pipeline = workers[sweep % max_workers];
-
-    auto dep = dlaf::internal::whenAllLift(sweep, w_pipeline(), deps[0]) |
-               dlaf::internal::transform(policy_hp, init_sweep) | ex::make_future();
-    copy_tridiag(sweep, std::move(dep));
-
-    const auto steps = nrStepsForSweep(sweep, size, b);
-    for (SizeType step = 0; step < steps; ++step) {
-      auto dep_index = std::min(step + 1, deps.size() - 1);
+  {
+    pika::scoped_annotation ann3("band_to_tridiag/sweeps");
+    const auto sweeps = nrSweeps<T>(size);
+    for (SizeType sweep = 0; sweep < sweeps; ++sweep) {
+      // Create the first max_workers workers and then reuse them.
+      auto& w_pipeline = workers[sweep % max_workers];
+      auto thread_hint = static_cast<std::remove_cv_t<decltype(num_threads)>>(sweep) % num_threads;
+      dlaf::internal::Policy<Backend::MC>
+          policy(pika::threads::thread_priority::high,
+                 pika::threads::thread_schedule_hint(pika::threads::thread_schedule_hint_mode::thread,
+                                                     thread_hint));
```

Comment:
Does this actually help? Non-exhaustive tests seem to say yes, but not significantly.

Reply:
Removed.